Add GCP KMS documentation (#369)
digital-asset/docs.daml.com: docs/2.7.0/docs/canton/architecture/security.rst

Short-term keys do not require additional protection because they are derived from a secret that is already
transmitted and stored in an encrypted form using a long-term public encryption key.

> Long-term keys should include an operational security policy that requires to rotate these keys periodically

Reviewer: ```suggestion
Long-term keys should be governed by an operational security policy that requires to rotate these keys periodically
```

   image taken from https://docs.aws.amazon.com/kms/latest/developerguide/rotate-keys.html

The manual rotation of a wrapper key requires not only the creation of a new KMS key but also the re-encryption of our data with it.
To do this Canton node administrators can request a :ref:`manual rotation of the KMS wrapper key through the Canton console <manual-kms-wrapper-key-rotation>`.

KMS Key Rotation
^^^^^^^^^^^^^^^^

When Canton's signing and encryption keys are off-sourced to a KMS (rather than encrypted at rest with a KMS wrapper key)
their rotation has to be operated manually. Neither AWS or GCP provide automatic asymmetric key rotation.

> Nevertheless, this can be easily achieved by requesting a :ref:`standard rotation of Canton's keys <rotating-canton-keys>`,

Reviewer 2: ```suggestion
Manual key rotation is achieved by requesting a :ref:`standard rotation of Canton's keys <rotating-canton-keys>`,
```

> Nevertheless, this can be easily achieved by requesting a :ref:`standard rotation of Canton's keys <rotating-canton-keys>`,

Reviewer: you should add the caveat that this only works when canton generates the new keys automatically, not when the new key has already been generated in the KMS (the GS custodian model). we need to document the latter too, can be in a separate PR if not already done.

> which in this particular case will also involve the rotation of the underlying KMS key.

Reviewer 2: ```suggestion
which in this particular case also involves the rotation of the underlying KMS key.
```
